### ISSUE TITLE
Support Terraform private registry

### ISSUE THREAD

**Issue by sebgamby** [coder/coder#8458]
Hello,

This is a feature request. My company is very security-sensitive and our policy is thus to use private registries. Furthermore those private registries use a server certificate issued from a self-signed CA certificate. The request would be to add options to "coder templates create" (and possible other relevant commands), along with server configuration items, that allow to:

- Provide an alternate registry URL for downloading the Terraform modules 
- Switch-off the TLS server verification
- Provide the path to a self-signed CA certificate for the TLS server verification

Thanks for your time.

**Comment by bpmct** [comment]
Hey, thanks for the feature request!

Right now you can support private registries, albeit it is a bit involved:

1) Modify Coder image to add your internal CAs
2) Edit the terraform.tfrc to change the default registry to the private registry

You can use our docs in https://coder.com/docs/v2/latest/install/offline + [Terraform's CLI configuration docs](https://developer.hashicorp.com/terraform/cli/config/config-file) as a pointer.

Let's keep this issue open for a bit because I totally agree there are things we can do in the product to make this easier.

**Comment by sebgamby** [comment]
Hello,

Sorry for the delay. I have tried to add our internal CA with the usual:
* COPY the the CRT file in /usr/local/share/ca-certificates
* RUN update-ca-certificates

It seems to work as I can execute "apt update" successfully when accessing the container. The Debian packages are downloaded through our company proxy which interrupts the TLS connection to run an antivirus validation. The server certificate for the TLS connection between the local client and the proxy is thus signed with our internal CA.

According to the GOLANG documentation, it should work but I still get "tls: failed to verify certificate: x509: certificate signed by unknown authority". I have tried use the environment variables SSL_CERT_FILE, SSL_CERT_DIR, insecureskipverify but could not get a different result. Am I missing something?
